src/corehost/cli/json_parser.h

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __JSON_PARSER_H__
+#define __JSON_PARSER_H__
+
+#include "pal.h"
+#include "rapidjson/document.h"
+#include "rapidjson/fwd.h"
+#include <memory>
+
+class json_parser_t {
+    public:
+        const rapidjson::Document& document() const { return m_document; }
+
+        bool parse_file(const pal::string_t& path)
+        {
+            pal::ifstream_t file{path};
+            return parse_stream(file, path);
+        }
+
+        bool parse_stream(pal::istream_t& stream, const pal::string_t& context);
+        bool parse_string(const pal::string_t& str, const pal::string_t& context);
+
+    private:
+        // JSON files parsed by this class are always encoded in UTF-8, so use char[]
+        // instead of pal::char_t[] for m_json.
+        std::unique_ptr<char[]> m_json;
```

Comment:
I didn't realize what you were doing here when I suggested this. In this case, using `std::vector<char>` might be easier since it provides many resize and manipulation functions.

Reply:
Done.